cpp/src/cylon/ops.cpp

```diff
@@ -0,0 +1,165 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "table.hpp"
+#include "ops.hpp"
+#include "net/mpi/mpi_operations.hpp"
+
+namespace cylon {
+
+/**
+ * A utility function for finding splits
+ * @param ctx
+ * @param left left table
+ * @param right right table
+ * @param left_column_indexes vector of column indexes we are working with
+ * @param right_column_indexes vector of column indexes we are working with
+ * @param left_splits return left splits
+ * @param right_splits return right splits
+ */
+Status Get_Splits(const std::shared_ptr <cylon::CylonContext> &ctx,
+                std::shared_ptr <cylon::Table> &left,
+                 std::shared_ptr <cylon::Table> &right,
+                 std::vector<int> left_column_indexes,
+                 std::vector<int> right_column_indexes,
+                 int &left_splits,
+                 int &right_splits) {
+  std::vector<int64_t> left_sizes = cylon::util::GetBytesAndElements(left->get_table(), left_column_indexes);
+  std::vector<int64_t> right_sizes = cylon::util::GetBytesAndElements(left->get_table(), right_column_indexes);
+  left_sizes.insert(left_sizes.end(), right_sizes.begin(), right_sizes.end());
+  std::vector<int64_t> totals(4);
```

Comment:
if these are fixed sizes, shall we use `std::array<int64_t, 4>` instead of `vector`?